.cirrus.yml

```diff
@@ -86,23 +86,29 @@ iox2_common_build_and_test_no_doc_tests_release_template: &IOX2_COMMON_BUILD_AND
 
 iox2_ffi_common_debug_template: &IOX2_FFI_COMMON_DEBUG
   ffi_script:
-    - cmake -S . -B target/ffi/build -DCMAKE_INSTALL_PREFIX=target/ffi/install -DBUILD_EXAMPLES=ON -DBUILD_TESTING=ON -DCMAKE_BUILD_TYPE=Debug
+    - ./internal/scripts/ci_build_and_install_iceoryx_hoofs.sh
+    - cmake -S . -B target/ffi/build -DCMAKE_BUILD_TYPE=Debug -DBUILD_EXAMPLES=ON -DBUILD_TESTING=ON -DCMAKE_INSTALL_PREFIX=target/ffi/install -DCMAKE_INSTALL_PREFIX="$( pwd )/target/iceoryx/install"
     - cmake --build target/ffi/build
     - cmake --install target/ffi/build
     - target/ffi/build/tests/iceoryx2-c-tests
     - rm -rf target/ffi/build
-    - cmake -S examples/c -B target/ffi/out-of-tree -DCMAKE_PREFIX_PATH=$( pwd )/target/ffi/install -DCMAKE_BUILD_TYPE=Debug
-    - cmake --build target/ffi/out-of-tree
+    - cmake -S examples/c -B target/ffi/out-of-tree-c -DCMAKE_BUILD_TYPE=Debug -DCMAKE_PREFIX_PATH="$( pwd )/target/ffi/install;$( pwd )/target/iceoryx/install"
+    - cmake --build target/ffi/out-of-tree-c
+    - cmake -S examples/cxx -B target/ffi/out-of-tree-cxx -DCMAKE_BUILD_TYPE=Debug -DCMAKE_PREFIX_PATH="$( pwd )/target/ffi/install;$( pwd )/target/iceoryx/install"
+    - cmake --build target/ffi/out-of-tree-cxx
```

Comment:
Could we add a clang-format and clang-tidy CI check right from the beginning?

Reply:
I moved that task to a follow up to not block the C++ binding work